src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -98,9 +98,19 @@ public void CheckTypesOnDeclaration(CheckTypesContext context, DType actualBodyR
             Contracts.AssertValue(actualBodyReturnType);
             Contracts.AssertValue(errorContainer);
 
-            if (!ReturnType.Kind.Equals(actualBodyReturnType.Kind) && !actualBodyReturnType.CoercesTo(ReturnType, true, false, context.Features.PowerFxV1CompatibilityRules))
+            if (!ReturnType.Kind.Equals(actualBodyReturnType.Kind))
```

Comment:
Why Equals instead of Accepts?

Wouldn't it be fine to return, e.g. a Record with fewer fields?

Reply:
Changed to Accepts()